test/system/250-systemd.bats

```diff
@@ -142,14 +142,21 @@ function service_cleanup() {
 @test "podman generate systemd - envar" {
     cname=$(random_string)
     FOO=value BAR=%s run_podman create --name $cname --env FOO -e BAR --env MYVAR=myval \
-        $IMAGE sh -c 'printenv && sleep 100'
+        $IMAGE sh -c 'printenv && echo READY; trap 'exit' SIGTERM; while :; do sleep 0.1; done'
 
     # Start systemd service to run this container
     service_setup
 
-    # Give container time to start; make sure output looks top-like
-    sleep 2
-    run_podman logs $cname
+    # Give container time to start and print output
+    local timeout=10
+    while [[ $timeout -ge 0 ]]; do
+        run_podman logs $cname
+        if [[ "$output" =~ READY ]]; then
+            break
+        fi
+        sleep 0.5
+    done
```

Comment:
This could just be `wait_for_ready $cname`, perhaps with a comment confirming that the only two `return`s from that helper are both immediately after `run_podman` which sets `$output`. Or, if you wish to keep this loop as a guarantee of `$output`, please decrement `$timeout` so the loop isn't infinite

Reply:
yeah using wait_for_ready is best

Reply:
done